lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -1454,3 +1455,27 @@ bool CGHeroInstance::hasVisions(const CGObjectInstance * target, const int subty
 
 	return (distance < visionsRange) && (target->pos.z == pos.z);
 }
+
+bool CGHeroInstance::isMissionCritical(const CMap* map) const
```

Comment:
Map objects should not use map pointers from outside. For now they should use IObjectInterface::cb for game state access.